### PR TITLE
Speed up Android & iOS fixture builds on CI

### PR DESCRIPTION
**Pull request by nickdowell**
## Goal

Reduce time for CI builds to complete

## Changeset

Speeds up Android & iOS fixture builds by avoiding re-building the Plugin and using `-ForceUnity` to speed up compilation.

Skips building the Plugin for engine versions 4.25, 4.26 and 5 on dev branches since we're much more likely to encounter issues with 4.23.

## Testing

Overall build times have been reduced from ~45 to ~30 minutes 🎉 

Note that we can still encounter slow builds if shaders need to be recompiled (Unreal Engine caches them between builds.)

### CODE REVIEW

**Review by kattrali — APPROVED** [review]

**Review by twometresteve — APPROVED** [review]
Made a few comments, but happy with it as it is.

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -21,6 +27,7 @@ steps:
     timeout_in_minutes: 30
 
   - name: 'Build Plugin - UE 4.25'
+    if: build.branch =~ /main|next|^release|^integration/ || build.message =~ /\[full ci\]/
```
[the remark this answers is not in the document]
Comment (nickdowell) [reply .buildkite/pipeline.yml:30]: Dynamically loading a separate pipeline file adds more complexity

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -73,42 +72,48 @@ steps:
   ##############################################################################
 
   - name: 'Build Android test fixtures - UE 4.27'
+    depends_on: [plugin_4_27]
```
[the remark this answers is not in the document]
Comment (nickdowell) [reply .buildkite/pipeline.yml:75]: Ah, I mistakenly thought depends_on needed to be an array

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -10,6 +6,16 @@ steps:
   #
   ##############################################################################
 
+  - name: 'Build Plugin - UE 4.27'
+    agents:
+      queue: opensource-mac-unreal-4.27
+    env:
+      UE_VERSION: "4.27"
+    command: make package
+    artifact_paths: [Build/Plugin/*.zip]
```
[the remark this answers is not in the document]
Comment (twometresteve) [reply .buildkite/pipeline.yml:15]: idm mind that much, sometimes people like consistency over succinctness.

File: .buildkite/pipeline.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (twometresteve) [reply .buildkite/pipeline.yml:30]: Yes, I do think the Cocoa and Android ones are too complex, but I think it's worked ok on Unity (for example) where we have just two levels of build.  I makes it easy to see what will get run on most pushes.

It also means we can more easily decide on the type of build based on what is being PRed into, not just the current branch (although dev/release processes may mean we don't need that).  I'd argue that repeating the condition in multiple steps is complexity in itself and liable to get out of step.

But I'm not proposing we change it here though, I can PR it at a later date and we can take a fresh look.

[the remark this answers is not in the document]
Comment (nickdowell) [reply .buildkite/pipeline.yml:30]: FWIW the amount of duplication in and overall size of the Cocoa pipelines bothers me.

Wonder if producing the pipeline programatically (e.g. from a Ruby script) instead would be easier to maintain 🤔

[the remark this answers is not in the document]
Comment (twometresteve) [reply .buildkite/pipeline.yml:30]: Agreed on both points.  We're dirtying the PR now, but very happy to continue the discussion IRL and get any tickets raised.

**Pull request merged**